### PR TITLE
Added Flush method send all events before app is exited

### PR DESCRIPTION
**Pull request by smurfpandey**
In my application errors were not getting reported. After calling flush method it started to work.

### CODE REVIEW

**Comment by onrik** [comment]
Nice idea. Please move timeout to Hook struct.

**Comment by smurfpandey** [comment]
Timeout is only needed for flush method. And if we move the timeout to struct, when will the user provide it's value?

**Comment by onrik** [comment]
```go
func (hook *Hook) SetFlushTimeout(timeout time.Duration) {
	hook.flushTimeout = timeout
}
```

File: sentry/sentry.go [hunk 1]
```diff
@@ -83,6 +86,14 @@ func (hook *Hook) SetEnvironment(environment string) {
 	hook.environment = environment
 }
 
+func (hook *Hook) SetFlushTimeout(timeout time.Duration) {
+	hook.flushTimeout = timeout
+}
+
+func (hook *Hook) Flush() {
+	hook.client.Flush(flushTimeout)
```
Comment (onrik) [inline sentry/sentry.go]: `hook.flushTimeout`

File: sentry/sentry.go [hunk 2]
```diff
@@ -20,12 +22,13 @@ var (
...
-	release     string
-	environment string
-	prefix      string
+	client       *sentry.Client
+	levels       []logrus.Level
+	tags         map[string]string
+	release      string
+	environment  string
+	prefix       string
+	flushTimeout time.Duration
```
Comment (onrik) [inline sentry/sentry.go:31]: Need a default value.

Comment (smurfpandey) [reply sentry/sentry.go:31]: Added 10sec timeout as default value in NewHook method.

File: sentry/sentry.go (hunk 1 above)
Comment (smurfpandey) [reply sentry/sentry.go]: Fixed

**Pull request merged**